e2eTests/scenarios/FederatedCredentials/runtest.ps1

```diff
@@ -0,0 +1,89 @@
+#
+# This test tests the following scenario:
+#
+# bcsamples-bingmaps.appsource is setup to use an Azure KeyVault for secrets and app signing
+# Access to the Azure KeyVault is using federated credentials (branches main and e2e)
+# bcsamples-bingmaps.appsource is using a Entra ID app registration for delivering to AppSource
+# The Entra ID app registration is using federated credentials (branches main and e2e)
+#
+#  - Remove branch e2e in repository microsoft/bcsamples-bingmaps.appsource (if it exists)
+#  - Create a new branch called e2e in repository microsoft/bcsamples-bingmaps.appsource (based on main)
+#  - Update AL-Go System Files in branch e2e in repository microsoft/bcsamples-bingmaps.appsource
+#  - Invoke CI/CD in branch e2e in repository microsoft/bcsamples-bingmaps.appsource
+#  - Check that artifacts are created and signed
+#  - Check that the app is delivered to AppSource
+#  - Remove the branch e2e in repository microsoft/bcsamples-bingmaps.appsource
+#
+'@
+
+$errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+
+Remove-Module e2eTestHelper -ErrorAction SilentlyContinue
+Import-Module (Join-Path $PSScriptRoot "..\..\e2eTestHelper.psm1") -DisableNameChecking
+
+$branch = "e2e"
+$template = "https://github.com/$appSourceTemplate"
+$repository = 'microsoft/bcsamples-bingmaps.appsource'
+
+SetTokenAndRepository -github:$github -githubOwner $githubOwner -token $token -repository $repository
+$headers = @{
+    "Authorization" = "token $token"
+    "X-GitHub-Api-Version" = "2022-11-28"
+    "Accept" = "application/vnd.github+json"
+}
+$existingBranch = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$repository/branches/$branch 2> $null | ConvertFrom-Json
+if ($existingBranch.PSObject.Properties.Name -eq 'Name' -and $existingBranch.Name -eq $branch) {
+    Write-Host "Removing existing branch $branch"
+    Invoke-RestMethod -Method Delete -Uri "https://api.github.com/repos/$repository/git/refs/heads/$branch" -Headers $headers
+    Start-Sleep -Seconds 10
+}
+$latestSha = (gh api /repos/$repository/commits/main | ConvertFrom-Json).sha
+gh api --method POST -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$repository/git/refs -f ref=refs/heads/$branch -f sha=$latestSha
+
+# Upgrade AL-Go System Files to test version
+RunUpdateAlGoSystemFiles -directCommit -wait -templateUrl $template -repository $repository -branch $branch | Out-Null
+
+# Run CI/CD workflow
+$run = RunCICD -repository $repository -branch $branch -wait
+
+# Check that workflow run uses federated credentials and signing was successful
+Test-LogContainsFromRun -repository $repository -runid $run.id -jobName 'Build Main App (Default)  Main App (Default)' -stepName 'Sign' -expectedText 'Connecting to Azure using clientId and federated token'
```

Comment:
For checking if the signing was successful, isn't it better to actually check if the files are signed, not just trust the logs?

Reply:
I could, but this end 2 end test tests that we can use federated credentials with a keyvault - it isn't really testing whether signing is successful. Then again - we don't have any end 2 end test that actually tests whether the signing is successful.

In order to check whether a file is signed, I would need to download the app, download the NAVSIP DLL and register that - and then test - will test this.

Reply:
Added a check for signature - running end 2 end to test it